osu.Game/Online/Multiplayer/StatefulMultiplayerClient.cs

```diff
@@ -178,6 +183,39 @@ public Task ChangeSettings(Optional<string> name = default, Optional<PlaylistIte
             });
         }
 
+        /// <summary>
+        /// Toggles the <see cref="LocalUser"/>'s ready state.
+        /// </summary>
+        /// <returns><c>true</c> if this toggle triggered a gameplay start; <c>false</c> otherwise.</returns>
+        /// <exception cref="InvalidOperationException">If a toggle of ready state is not valid at this time.</exception>
+        public async Task<bool> ToggleReady()
+        {
+            var localUser = LocalUser;
+
+            if (localUser == null)
+                return false;
+
+            switch (localUser.State)
+            {
+                case MultiplayerUserState.Idle:
+                    await ChangeState(MultiplayerUserState.Ready);
+                    return false;
+
+                case MultiplayerUserState.Ready:
+                    if (Room?.Host?.Equals(localUser) == true)
```

Comment:
hmm, not 100% keen on this existing here with the current method name (and the xmldoc not really mentioning *when* a game start is triggered).

Reply:
Hm, good point. *Probably* saner to move this to the action itself I suppose (and duplicate in test for testing)...

How does something like this look?

```csharp
                OnReadyClick = async () =>
                {
                    readyClickedOperation = ongoingOperationTracker.BeginOperation();

					if (Client.IsHost && Client.LocalUser?.State == MultiplayerUserState.Ready)
					{
						await Client.StartGameplay();
						return;
					}

                    await Client.ToggleReady() // <- does just the toggle logic
					readyClickedOperation.Dispose();
					readyClickedOperation = null;
                }
```

(that's with surrounding names removed because those didn't make much sense in retrospect, either).

Reply:
looks better, for sure.